_posts/2018-02-13-the-pros-and-cons-of-state-management.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title: "The Pros and Cons of State Management"
+date: 2018-02-13 09:11:01
+tags:
+- Web
+- Vue
+- Vuex
+- JavaScript
+- StateManagement
+twitter_text: "The Pros and Cons of #StateManagement"
+authors: Jaime Jones
+---
+
+Let's talk about state management for a minute. There's recently been a bunch of hubub about Redux on Twitter, and arguments about the usefulness of state management. Here at Ritter we haven't utilized Redux specifically, but we frequently leverage Vue's own state management library, Vuex.
+
+I'm going to do a brief overview of Vuex and then talk about the pros and cons of state management. There's something that I will point out right now, and reiterate later: State management is a very useful paradigm, but it is _not_ intended for use in every single project.
+
+## What is Vuex?
+I'm going to do a quick summary of Vuex and how it works, but I recommend that you check out the [Vuex docs](https://vuex.vuejs.org/en/intro.html) if you are interested or just looking for a more in depth explanation.
+
+Vuex is a state management library, aimed at creating a centralized source that can drive the rest of your application. It is composed of three main parts: State, Mutations, and Actions. For the sake of this explanation, we will use an example of a blog which must fetch the posts to display.
+
+State houses the data that lives in your application, and can be referred to as your 'single source of truth.' No matter where in your application you access data that is housed in the state, it will remain the same across each application because it is coming from this single source.
+```
+state: {
+  posts: []
+}
+```
+
+Mutations are methods where we actually modify the data properties on the state. These mutations can be accessed from anywhere within your application, but they serve as a way of interfacing with the state through your application. Mutations are called through `commit`, and must take in the state so that it can modify it, and can optionally take in a payload.
+```
+mutations: {
+  setPosts: (state, payload) => {
+    state.posts = payload;
+  }
+}
+```
+
+Actions are methods that trigger mutations, and actions are also where you would house any asynchronous operations. Actions are called through `dispatch`. Actions can take in the different methods that it can use (such as `commit`, or `dispatch`, and the `state`) and an optional payload.
+```
+actions: {
```

Comment:
If each code block had an indication of the filename or something (perhaps in a comment at the first line, or as a header) would that help readers understand how these code blocks are organized in the project?

Reply:
These are usually all held in the `store.js`

Reply:
Oh, the post does say that... :smiley: